File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        ar, ac = [0] * m, [0] * n
        for i in range(0, m):
            for j in range(0, n):
                if mat[i][j] == 1:
                    ar[i] += 1
                    ac[j] += 1
        
        ans = 0
        for i in range(0, m):
            for j in range(0, n):
                if mat[i][j] == 1 and ar[i] == 1 and ac[j] == 1: ans += 1

        return ans

# array, matrix
# time: O(mn)
# space: O(m + n)

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.go
func numSpecial(mat [][]int) int {
    m, n := len(mat), len(mat[0])
    ar, ac := make([]int, m), make([]int, n) // freq row and col
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if mat[i][j] == 1 {
                ar[i]++
                ac[j]++
            }
        }
    }

    ans := 0
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if mat[i][j] == 1 && ar[i] == 1 && ac[j] == 1 { ans++ }
        }
    }

    return ans
}

// array, matrix
// time: O(mn)
// space: O(m + n)